src/Illuminate/Queue/QueueManager.php

```diff
@@ -110,6 +120,30 @@ public function stopping($callback)
         $this->app['events']->listen(Events\WorkerStopping::class, $callback);
     }
 
+    /**
+     * Add metadata to the payload that will be queued.
+     *
+     * @param  string|array|null  $key
+     * @param  mixed  $value
+     * @return  self|\Illuminate\Support\Collection
+     */
+    public function metadata($key, $value = null)
+    {
+        if (is_null($key)) {
+            return $this->metadata;
+        }
+
+        if (is_array($key)) {
+            foreach ($key as $itemKey => $itemData) {
+                $this->metadata($itemKey, $itemData);
+            }
```

Comment:
Would a merge work?

```php
$this->metadata = $this->metadata->merge($key);
```

Reply:
Yeah that would be a better solution. Will update later